src/DateTimeParser.h

```diff
@@ -79,19 +79,18 @@ class DateTimeParser {
 
   // A flexible time parser for the most common formats
   bool parseTime() {
-    if (!consumeInteger(2, &hour_))
-      return false;
-    consumeThisChar(':');
-    if (!consumeInteger(2, &min_))
-      return false;
-    consumeThisChar(':');
-    consumeSeconds(&sec_, NULL);
+    bool valid = consumeInteger(2, &hour_) &&
```

Comment:
Do you mind sticking with the previous structure? i.e. `return false` when a condition isn't met - I find that easier to understand.